src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,54 @@
-'use strict';
-
 class Animal {
-  // write your code here
+  static alive = [];
+
+  constructor(name) {
+    this.name = name;
+    this.health = 100;
+    this.hidden = false;
+    Animal.alive.push(this);
+  }
+
+  checkHealth() {
+    if (this.health <= 0) {
+      this.die();
+    }
+  }
+
+  die() {
+    Animal.alive = Animal.alive.filter((animal) => animal !== this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name) {
+    super('name');
```

Comment:
[CODE QUALITY ISSUE] - In the `super()` call, you should pass the `name` argument from the constructor, not a string 'name'. The same issue is on line 38.